-code-checks

ERROR: Build errors found during clang-tidy run.
```
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:228:9: error: use of undeclared identifier 'endVertex' [clang-diagnostic-error]
    if (endVertex && endVertex->position().perp2() * lengthUnitConversionFactor2_ < beamPipeRadius2_) {
        ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:228:22: error: use of undeclared identifier 'endVertex' [clang-diagnostic-error]
    if (endVertex && endVertex->position().perp2() * lengthUnitConversionFactor2_ < beamPipeRadius2_) {
                     ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:229:7: error: 'continue' statement not in loop statement [clang-diagnostic-error]
      continue;
      ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:234:22: error: use of undeclared identifier 'particle' [clang-diagnostic-error]
        new Particle(particle.pdg_id(),
                     ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:235:46: error: use of undeclared identifier 'productionVertex' [clang-diagnostic-error]
                     math::XYZTLorentzVector(productionVertex->position().x() * lengthUnitConversionFactor_,
                                             ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:236:46: error: use of undeclared identifier 'productionVertex' [clang-diagnostic-error]
                                             productionVertex->position().y() * lengthUnitConversionFactor_,
                                             ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:237:46: error: use of undeclared identifier 'productionVertex' [clang-diagnostic-error]
                                             productionVertex->position().z() * lengthUnitConversionFactor_,
                                             ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:238:46: error: use of undeclared identifier 'productionVertex' [clang-diagnostic-error]
                                             productionVertex->position().t() * timeUnitConversionFactor_),
                                             ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:239:46: error: use of undeclared identifier 'particle' [clang-diagnostic-error]
                     math::XYZTLorentzVector(particle.momentum().x() * momentumUnitConversionFactor_,
                                             ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:240:46: error: use of undeclared identifier 'particle' [clang-diagnostic-error]
                                             particle.momentum().y() * momentumUnitConversionFactor_,
                                             ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:241:46: error: use of undeclared identifier 'particle' [clang-diagnostic-error]
                                             particle.momentum().z() * momentumUnitConversionFactor_,
                                             ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:242:46: error: use of undeclared identifier 'particle' [clang-diagnostic-error]
                                             particle.momentum().e() * momentumUnitConversionFactor_)));
                                             ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:244:18: error: use of undeclared identifier 'exoticRelativeId' [clang-diagnostic-error]
    if (isExotic(exoticRelativeId)) {
                 ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:245:35: error: use of undeclared identifier 'exoticRelativeId' [clang-diagnostic-error]
      newParticle->setMotherPdgId(exoticRelativeId);
                                  ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:249:9: error: unknown type name 'endVertex'; did you mean 'SimVertex'? [clang-diagnostic-error]
    if (endVertex) {
        ^~~~~~~~~
        SimVertex
--
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:249:18: error: expected unqualified-id [clang-diagnostic-error]
    if (endVertex) {
                 ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:251:12: error: use of undeclared identifier 'endVertex' [clang-diagnostic-error]
          (endVertex->position().t() - productionVertex->position().t()) * timeUnitConversionFactor_;
           ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:251:40: error: use of undeclared identifier 'productionVertex' [clang-diagnostic-error]
          (endVertex->position().t() - productionVertex->position().t()) * timeUnitConversionFactor_;
                                       ^
FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc:278:3: error: expected unqualified-id [clang-diagnostic-error]
  return std::unique_ptr<Particle>();
  ^
Suppressed 659 warnings (657 in non-user code, 2 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:128: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/FastSimulation/SimplifiedGeometryPropagator/src/Decayer.cc
+++ b/FastSimulation/SimplifiedGeometryPropagator/src/Decayer.cc
@@ -37,7 +37,7 @@ void fastsim::Decayer::decay(const Particle& particle,
   // snip decay products of exotic particles or their children. These decay products are preserved from the event record.
   // limitation: if exotic incurs heavy energy loss during propagation, the saved decay products could be too hard.
 
-if (isExotic(pid) || isExotic(particle.getMotherPdgId())){
+if (isExotic(pid) || isExotic(particle.getMotherPdgId())) {
     return;
 }
 

--- a/FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc
+++ b/FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc
@@ -219,7 +219,7 @@ std::unique_ptr<fastsim::Particle> fastsim::ParticleManager::nextGenParticle() {
     if (productionVertex->position().perp2() * lengthUnitConversionFactor2_ > beamPipeRadius2_)  //
     {
       exoticRelativesChecker(productionVertex, exoticRelativeId, 0);
-    if (!isExotic(exoticRelativeId)){
+      if (!isExotic(exoticRelativeId)) {
         continue;
       }
     }
@@ -281,7 +281,7 @@ std::unique_ptr<fastsim::Particle> fastsim::ParticleManager::nextGenParticle() {
 void fastsim::ParticleManager::exoticRelativesChecker(const HepMC::GenVertex* originVertex,
                                                       int& exoticRelativeId_,
                                                       int ngendepth = 0) {
-   if (ngendepth > 99 || exoticRelativeId_ == -1 || isExotic(std::abs(exoticRelativeId_))) 
+  if (ngendepth > 99 || exoticRelativeId_ == -1 || isExotic(std::abs(exoticRelativeId_))) 
     return;
   ngendepth += 1;
   std::vector<HepMC::GenParticle*>::const_iterator relativesIterator_ = originVertex->particles_in_const_begin();

--- a/FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc
+++ b/FastSimulation/SimplifiedGeometryPropagator/src/ParticleManager.cc
@@ -219,7 +219,7 @@ std::unique_ptr<fastsim::Particle> fastsim::ParticleManager::nextGenParticle() {
     if (productionVertex->position().perp2() * lengthUnitConversionFactor2_ > beamPipeRadius2_)  //
     {
       exoticRelativesChecker(productionVertex, exoticRelativeId, 0);
-if (!isExotic(exoticRelativeId))
+    if (!isExotic(exoticRelativeId)){
         continue;
       }
     }

